src/services/pagination/LengthAwarePaginator.js

```diff
@@ -0,0 +1,81 @@
+import Paginator from './Paginator';
+import LesgoException from '../../exceptions/LesgoException';
+
+const FILE = 'Services/pagination/LengthAwarePaginator';
+
+export default class LengthAwarePaginator extends Paginator {
+  /**
+   * Constructor
+   *
+   * @param db
+   * @param sql
+   * @param sqlParams
+   * @param perPage
+   * @param currentPage
+   * @param total
+   */
+  constructor(db, sql, sqlParams, perPage = 10, currentPage = 1, total = null) {
+    if (total !== null && typeof total !== 'number') {
+      throw new LesgoException(
+        "Invalid type for 'total'",
+        `${FILE}::INVALID_TYPE_TOTAL`,
+        500,
+        { perPage }
+      );
+    }
+    super(db, sql, sqlParams, perPage, currentPage);
+    this.totalProp = total;
+  }
+
+  /**
+   * Get the last page.
+   *
+   * @returns {Promise<number>}
+   */
+  async lastPage() {
+    return this.calculateTotalNumberOfPages();
+  }
+
+  /**
+   * Total items in all pages.
+   *
+   * @returns {null|number}
+   */
+  async total() {
+    if (this.totalProp === null) {
+      this.totalProp = await this.countTotalItems();
+    }
+
+    return this.totalProp;
+  }
+
+  async toObject() {
+    const obj = await super.toObject();
+    const { items } = obj;
+    delete obj.items;
+
+    return {
+      ...obj,
+      last_page: await this.lastPage(),
+      total: await this.total(),
+      items,
+    };
+  }
+
+  // They act as protected methods.
+
+  /**
+   * Count total items with basic implementation.
+   *
+   * @returns {Promise<number>}
+   */
+  async countTotalItems() {
+    const resp = await this.dbProp.select(this.sqlProp, this.sqlParamsProp);
```

Comment:
I don't know the exact function of this but it looks like an SQL query to me, will it be possible just to return the count output from SQL rather than this?

Reply:
Because it would be hard to parse the supplied SQL query string to include the `count()` function in it, as that query will be used to get the actual data as well. The value of the `this.sqlProp` in this context would be similar to `SELECT table_1, table_2 FROM sample_table WHERE id = 1`.

Reply:
And I don't see a method in the `AuroraDBService` that would return the count of a query.